rerun-io/rerun
# Documentation search index

### What

- [x] Make it easy to setup a local database for testing purposes
  - `pixi run meilisearch-test-db`
- [x] Build a search index from documentation markdown files (no API references)
  - Queries can be run against it via the `cargo run -p re_build_search_index` REPL
- [x] Figure out a way to ingest public APIs in every language

The search index build should be added to the release process, but in a different PR.

How to run this locally:
```
pixi run meilisearch-test-db
pixi run cargo run --release -p re_build_search_index -- repl temp --url "http://localhost:7700" --master-key "test"
```

This will open up a REPL. Typing in a query and pressing enter will display the results. Close with CTRL+C or by using the `quit` command.

To test this on `landing`:
- `pixi run meilisearch-test-db`
- `git checkout` this PR: https://github.com/rerun-io/landing/pull/628
- In `.env.local`:
  - Make sure `NEXT_PUBLIC_MEILISEARCH_API_KEY` and `NEXT_PUBLIC_MEILISEARCH_URL` are _not_ set to anything (you can comment them out using `#`)
- Run `yarn dev` and open `http://localhost:3000`

### Checklist
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4780/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4780/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4780/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4780)
- [Docs preview](https://rerun.io/preview/8fb347606308cfa059cdb1df84118f69b5a92136/docs) 
- [Examples preview](https://rerun.io/preview/8fb347606308cfa059cdb1df84118f69b5a92136/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)
